chaos_genius/core/utils/data_loader.py

```diff
@@ -178,17 +177,30 @@ def _run_query(self, query):
     def _preprocess_df(self, df):
         df[self.dt_col] = pd.to_datetime(df[self.dt_col])
 
-        if self.kpi_info.get("timezone_aware"):
-            # if tz aware data, convert to given timezone
-            # and then strip tz information
+        # TODO: use the timezone_aware column in kpi table once updated
+        # tz-naive timestamps get localized to their database timezone.
+        if df[self.dt_col].dt.tz is None:
+            df[self.dt_col] = df[self.dt_col].dt.tz_localize(
+                DataSource.get_by_id(
+                    self.kpi_info["data_source"]
+                    ).as_dict["database_timezone"])
```

Comment:
The DataSource is already queried in the constructor:
```suggestion
                self.connection_info["database_timezone"])
```